src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -127,6 +127,15 @@ def dump_requirements(builder, interpreter, reqs, log, platforms=None):
       locations.add(dist.location)
 
 
+def subsystems_used():
+  """The subsystems used by the utility functions.
+
+  This function is provided so that consuming tasks can call it in their
+  subsystem_dependencies implementations.
+  """
+  return (PythonRepos, PythonSetup)
```

Comment:
This also points wrapping the utlity functions that need these into a class whose constructor requires PythonRepos & PythonSetup instances be passed in with no defaulting. That would prevent the problem going forward.

Reply:
That makes sense to me. I don't think it'd be too much work to update this PR with that.